### PR TITLE
Extract the Foundation from BCF.

### PR DESCRIPTION
**Pull request by ykulbak**
Add a reference to the Open CDE Foundation API specification.

Leave the Authorization section in BCF, for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,29 +4,22 @@
 # BCF REST API
 ![](https://raw.githubusercontent.com/BuildingSMART/BCF/master/Icons/BCFicon128.png)
 
-**Version 2.1** based on BCFv2.1.
+**Version 2.2** based on BCFv2.1.
```
Comment (pasi-paasiala) [inline README.md:7]: Should this be BCFv2.2?

File: README.md [hunk 2]
```diff
@@ -4,29 +4,22 @@
...
 [BCF XML GitHub repository](https://github.com/BuildingSMART/BCF)
 
+[Version 1.0 of the Open CDE Foundation API specification is available here](https://github.com/BuildingSMART/shared-common-API/tree/v1.0)
```
Comment (pasi-paasiala) [inline README.md:10]: When rendered, these don't seem to have line breaks. Try adding extra linefeeds between these lines.

File: README.md [hunk 3]
```diff
@@ -118,100 +111,35 @@
 
 # 1. Introduction
 
-BCF is a format for managing issues on a BIM project. The BCF-API supports the exchange of BCF issues between software applications via a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) web interface, which means that data is exchanged via Url-encoded query parameters and Json bodies over the Http protocol. Every resource described in this API has a corresponding Json schema (schema version draft-03).
+BCF is an open standard for managing issues on a BIM project. The BCF-API supports the exchange of BCF issues between software applications via a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) web interface, which means that data is exchanged via Url-encoded query parameters and Json bodies over the Http protocol. Every resource described in this API has a corresponding Json schema (schema version draft-03).
```
Comment (pasi-paasiala) [inline README.md:114]: URL instead of Url in "Url-encoded"?

File: README.md [hunk 4]
```diff
@@ -118,100 +111,35 @@
...
 An example of a client implementation in C# can be found here:
 [https://github.com/rvestvik/BcfApiExampleClient](https://github.com/rvestvik/BcfApiExampleClient)
 
-## 1.1 Paging, Sorting and Filtering
+## 1.1 Open CDE Foundation API
 
-When requesting collections of items, the BCF-API should offer URL parameters according to the OData v4 specification. It can be found at [http://www.odata.org/documentation/](http://www.odata.org/documentation/).
+BCF API is a member of the Open CDE API family. All Open CDE APIs are united by a shared common API called [Open CDE Foundation API](https://github.com/buildingSMART/shared-common-API). 
+The foundation API specifies a small number of services and a few conventions that are common to all Open CDE APIs. All BCF API implementations must implement the Foundation API and follow its conventions and guidelines. 
+Implementers should start by implementing the Founation API and only then continue to implement the BCF API.
```
Comment (pasi-paasiala) [inline README.md:126]: Typo in "Founation"

File: README.md (hunk 1 above)
Comment (ykulbak) [reply README.md:7]: Thanks, changed

File: README.md (hunk 3 above)
Comment (ykulbak) [reply README.md:114]: Rephrased the paragraph.

File: README.md (hunk 4 above)
Comment (ykulbak) [reply README.md:126]: cheers

File: README.md (hunk 2 above)
Comment (ykulbak) [reply README.md:10]: Thanks.

File: README.md [hunk 5]
```diff
@@ -118,100 +114,35 @@
 
 # 1. Introduction
 
-BCF is a format for managing issues on a BIM project. The BCF-API supports the exchange of BCF issues between software applications via a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) web interface, which means that data is exchanged via Url-encoded query parameters and Json bodies over the Http protocol. Every resource described in this API has a corresponding Json schema (schema version draft-03).
-Url schemas in this readme are relational to the BCF base servers Url if no absolute values are provided.
+BCF is an open standard for managing issues on a BIM project. The BCF-API supports the exchange of BCF issues between software applications via a [RESTful](https://en.wikipedia.org/wiki/Representational_state_transfer) web interface, which means that data is exchanged via HTTP query parameters and Json bodies. Every resource described in this API has a corresponding Json schema (schema version draft-03).
```
Comment (leosobral) [inline README.md:117]: Replace Json -> JSON everywhere

File: README.md [hunk 6]
```diff
@@ -118,100 +114,35 @@
...
+Url schemas in this README are relative to the BCF servers' base Url unless absolute values are provided.
 
 For security reasons, all API Http traffic should be sent via TLS/SSL over Https connection. Clients and Servers should both enforce secure connections and disallow unsafe connections.
```
Comment (leosobral) [inline README.md:120]: This paragraph about http traffic I think is better suited for foundations. 

Another comment, do we propose any miminum version of TLS? From a security perspective > 1.2 is recommended these days (can this be included for the foundations 1.0 release).

File: README.md [hunk 7]
```diff
@@ -118,100 +114,35 @@
...
-## 1.5 Http Status Codes
-
-The BCF API relies on the regular Http Status Code definitions. Good sources are [Wikipedia](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) or the [HTTP/1.1 Specification](https://tools.ietf.org/html/rfc7231).
+## 1.1 Open CDE Foundation API
 
-Generally, these response codes shall be used in the API:
-* `200 - OK` for `GET` requests that return data or `PUT` requests that update data
-* `201 - Created` for `POST` requests that create data
+BCF API is a member of the Open CDE API family. All Open CDE APIs are united by a shared common API called [Open CDE Foundation API](https://github.com/buildingSMART/shared-common-API). 
+The foundation API specifies a small number of services and a few conventions that are common to all Open CDE APIs. All BCF API implementations must implement the Foundation API and follow its conventions and guidelines.
```
Comment (leosobral) [inline README.md:128]: Should we add a note the Foundation 1.0 was released together with BCF 2.2 to make it clearer for new people implementing these standards?

**Review by leosobral — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -118,100 +114,35 @@
...
-
-## 1.8 Authorization
+## 1.2 Authorization
 
 API implementors can optionally choose to restrict the actions a user is allowed to perform on the BCF entities
 via the API. The global default authorizations for all entities are expressed in the project extensions schema and can
 be locally overridden in the entities themselves.
 
-### 1.8.1 Per-Entity Authorization
+### 1.2.1 Per-Entity Authorization
```
Comment (pbuts) [inline README.md:138]: There was a proposal to always include authorizations, so this might be obsolete

File: README.md [hunk 9]
```diff
@@ -4,29 +4,25 @@
...
-  * [1.4 Cross Origin Resource Sharing (CORS)](#14-cross-origin-resource-sharing-cors)
-  * [1.5 Http Status Codes](#15-http-status-codes)
-  * [1.6 Error Response Body Format](#16-error-response-body-format)
-  * [1.7 DateTime Format](#17-datetime-format)
-  * [1.8 Authorization](#18-authorization)
-    + [1.8.1 Per-Entity Authorization](#181-per-entity-authorization)
-    + [1.8.2 Determining Authorized Entity Actions](#182-determining-authorized-entity-actions)
-  * [1.9 Additional Response and Request Object Properties](#19-additional-response-and-request-object-properties)
-  * [1.10 Binary File Uploads](#110-binary-file-uploads)
-  * [1.11 Differences Between null and Empty Lists](#111-differences-between-null-and-empty-lists)
```
Comment (leosobral) [inline README.md:29]: Also the whole section #3 should be removed here in the index

File: README.md (hunk 8 above)
Comment (ykulbak) [reply README.md:138]: @pbuts that proposal was rejected

File: README.md (hunk 9 above)
Comment (ykulbak) [reply README.md:29]: Thanks, a huge miss.

File: README.md (hunk 5 above)
Comment (ykulbak) [reply README.md:117]: done

File: README.md (hunk 6 above)
Comment (ykulbak) [reply README.md:112]: Resolved in the group to leave it here and also make a similar note in Foundations. 

Also add a recommendation in foundation around the TLS protocol.

File: README.md (hunk 7 above)
Comment (ykulbak) [reply README.md:120]: Resolved to suffice with a mention in the release notes.

File: README.md (hunk 6 above)
Comment (ykulbak) [reply README.md:112]: https://github.com/buildingSMART/opencde-foundation-API/pull/14

**Comment by ykulbak** [comment]
@leosobral please review again, thanks

**Review by leosobral — APPROVED** [review]

**Review by jasollien — APPROVED** [review]

**Pull request merged**